Add Health and Fuel

zehuac2 opened the pull request:
Add Health and Fuel Bar prefabs, and three event handlers for linking the PlayerController class with the health and fuel bar.

zehuac2 committed:
Add Health and Fuel

Add Health and Fuel Bar prefabs, and three event handlers for linking the PlayerController class with the health and fuel bar.

Zehua-Chen commented:
**I think I may have conveyed the purpose of WillBeRespawnedEventHandler incorrectly**. What I intended for it was:

1. When the player is dead, the PlayerController.cs re-create the player;
2. After re-creating the player, the PlayerController.cs will then pass the newly-created GameObject to the WillBeRespawnedEventHandler, where I will rebuild the connection between the Fuel Bar and the Health Bar and the PlayerController class.

~~~
GameObject player = Instantiate(playerPrefab, position, rotation);
WillBeRespawned(player);
~~~
In the BarController.cs. I will reconnect the PlayerController reference, which I use to receive notifications on changes on Fuel and Health value

~~~
playerController = playerPrefab.GetComponent<PlayerController>();

//Somewhere else
playerController.HealthChanged += xxx;
playerController.FuelChanged += xxx;
~~~

Zehua-Chen commented:
Do not merge, bugs found!

zehuac2 committed:
Rework

I redo the health bar and menu bar connection thing.

zehuac2 commented:
In order for the PlayerController to interact with UI, do the following:
1. Make sure PlayerController directly or indirectly inherits from PlayerWithStatus class;
2. Link the Fuel Bar and Oil Bar Object with the designated slots of PlayerController in the editor;
3. Change health and fuel using the Health and Fuel **property** of the PlayerWithStatus class;

**A more detailed explanation can be found at [here on google drive.](https://drive.google.com/file/d/0B8geSOjLIVESSUNXR0FCZlJadFk/view)**

zehuac2 committed:
Fix

Instead of using gameobject, use string to reference the fuel and health bar object.
